controller/Security.php

```diff
@@ -39,8 +47,74 @@ public function jwks(): void
         $this->setResponse($response);
     }
 
+    #
+    # @TODO Test method, will be removed...
+    #
+    public function launch(): void
+    {
+        /** @var Lti1p3RegistrationRepository $registrationRepository */
+        $registrationRepository = $this->getServiceLocator()->get(Lti1p3RegistrationRepository::class);
+        $registration = $registrationRepository->find('registrationIdentifier');
+
+        # @TODO Provide a real userId here
+        $userIdentity = 'https://test-tao-deploy.docker.localhost/ontologies/tao.rdf#i5ef9f69837ace6f100dc57beb1439e';
+
+        $user = new UserIdentity(
+            $userIdentity,
+            'gabriel',
+            'gabriel@gabriel.com'
+        );
+
+        $ltiLaunchRequest = (new LtiLaunchRequestBuilder())->buildUserResourceLinkLtiLaunchRequest(
+            new ResourceLink('identifier'),
+            $registration, // $this->repository->find('local'),
+            $user,
+            null,
+            [
+                'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner'
+            ],
+            [
+                new ContextClaim('contextId'),
+                'myCustomClaim' => 'myCustomValue'
+            ]
+        );
+
+        $oidcLtiLaunchRequest = (new OidcLaunchRequestBuilder())->buildResourceLinkOidcLaunchRequest(
+            new ResourceLink('identifier'),
+            $registration,
+            $userIdentity,
+            null,
+            [
+                'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner'
+            ],
+            [
+                new ContextClaim('contextId'),
+                'myCustomClaim' => 'myCustomValue'
+            ]
+        );
```

Comment:
You could do something similar here:

```suggestion
        $oidcLtiLaunchRequest = $this->buildOidcLaunchRequest($registration, $userIdentity);
```

Reply:
The method `launch` is just to help you guys to test. It will be removed as soon the PR is approved :)